core/composio/sdk/shared_utils.py

```diff
@@ -0,0 +1,205 @@
+    KeyError: Missing 'type' in property definitions.
+    ValueError: Invalid 'type' for property or recursive model creation.
+
+    Note:
+    Requires global `schema_type_python_type_dict` for type mapping and `fallback_values` for default values.
+    """
+    required_fields = {}
+    optional_fields = {}
+    param_title = param_schema['title'].replace(" ", "")
+    required_props = param_schema.get('required', [])
+    # schema_params_object = param_schema.get('properties', {})
+    for prop_name, prop_info in param_schema.get('properties', {}).items():
+        prop_type = prop_info["type"]
+        prop_title = prop_info['title'].replace(" ", "")
+        prop_default = prop_info.get('default', fallback_values[prop_type])
+        if prop_type in schema_type_python_type_dict:
+            signature_prop_type = schema_type_python_type_dict[prop_type]
+        else:
+            signature_prop_type = pydantic_model_from_param_schema(prop_info)
+
+        if prop_name in required_props:
+            required_fields[prop_name] = (signature_prop_type, 
+                                Field(..., 
+                                    title=prop_title, 
+                                    description=prop_info.get('description', 
+                                                              prop_info.get('desc', 
+                                                                             prop_title))
+                                    ))
+        else:
+            optional_fields[prop_name] = (signature_prop_type, 
+                                Field(title=prop_title, 
+                                    default=prop_default
+                                    ))
+    fieldModel = create_model(param_title, **required_fields, **optional_fields)
+    return fieldModel
+        
+        
+def get_signature_format_from_schema_params(
+        schema_params
+):
+    """
+    Converts schema parameters into a list of `Parameter` objects for function signatures.
+
+    Parameters:
+    - schema_params (dict): Contains 'required' (list of names) and 'properties' (name-schema pairs).
+
+    Returns:
+    - List[Parameter]: Signature parameters, with required ones first.
+    """
+    required_parameters = []
+    optional_parameters = []
+
+    required_params = schema_params.get('required', [])
+    schema_params_object = schema_params.get('properties', {})
+    for param_name, param_schema in schema_params_object.items():
+        param_type = param_schema['type']
+        param_title = param_schema['title'].replace(" ", "")
+
+        if param_type in schema_type_python_type_dict:
+            signature_param_type = schema_type_python_type_dict[param_type]
+        else:
+            signature_param_type = pydantic_model_from_param_schema(param_schema)
+
+        param_default = param_schema.get('default', fallback_values[param_type])
+        param_annotation = Annotated[signature_param_type, param_schema.get('description', 
```

Comment:
Changed this function in master. Please have a look and update

Reply:
@sawradip is this resolved?

Reply:
Yes, it is not resolved. Could you check @sawradip

Reply:
Fixed.